src/graph.rs

```diff
@@ -446,6 +466,16 @@ impl Value {
         }
     }
 
+    pub fn as_string(&self, graph: &Graph) -> Result<&String, ExecutionError> {
```

Comment:
And I think it's more idiomatic for the return type here to be `&str` instead of `&String`